src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -1195,15 +1196,66 @@ public void testJdbc41CallableStatementMethods() throws Exception {
                     assertEquals("2017-05-19 10:47:15.1234567 +02:00",
                             cstmt.getObject("col14Value", microsoft.sql.DateTimeOffset.class).toString());
 
-                    // BigDecimal#equals considers the number of decimal places (OutParams always return 4 decimal
-                    // digits rounded up)
-                    assertEquals(0, cstmt.getObject(15, BigDecimal.class).compareTo(new BigDecimal("0.1235")));
+                    // BigDecimal#equals considers the number of decimal places (OutParams always return full precision as specified in the DB schema)
+                    assertEquals(0, cstmt.getObject(15, BigDecimal.class).compareTo(new BigDecimal("0.123456789")));
                     assertEquals(0,
-                            cstmt.getObject("col15Value", BigDecimal.class).compareTo(new BigDecimal("0.1235")));
+                            cstmt.getObject("col15Value", BigDecimal.class).compareTo(new BigDecimal("0.123456789")));
 
-                    assertEquals(0, cstmt.getObject(16, BigDecimal.class).compareTo(new BigDecimal("0.1235")));
+                    assertEquals(0, cstmt.getObject(16, BigDecimal.class).compareTo(new BigDecimal("0.1234567890123456789012345678901234567")));
                     assertEquals(0,
-                            cstmt.getObject("col16Value", BigDecimal.class).compareTo(new BigDecimal("0.1235")));
+                            cstmt.getObject("col16Value", BigDecimal.class).compareTo(new BigDecimal("0.1234567890123456789012345678901234567")));
+                }
+            }
+        }
+
+        @Test
+        public void testBigDecimalPrecision() throws SQLException {
+            try (Connection connection = getConnection()) {
+                String createProceduresSQL = "create procedure test_bigdecimal_3\n" +
+                "    @big_decimal_type      decimal(15, 3),\n" +
+                "    @big_decimal_type_o    decimal(15, 3) output\n" +
+                "as\n" +
+                "begin\n" +
+                "    set @big_decimal_type_o = @big_decimal_type;\n" +
+                "end;\n";
```

Comment:
We should move the create and drop of the procedure to the `init()` and `terminate()`. If the test fails for some reason mid run, then the drop call doesn't get executed, and then next run the create fails since the proc already exists. And, the procedure name should be prefixed/suffixed with a unique identifier eg. using `AbstractSQLGenerator.escapeIdentifier`. The create proc might fail if the proc name isn't unique, if multiple runs of this test is running against the same server.

Reply:
also, pls use code formatter this doesn't look formatted

Reply:
Updated the test case, please review.